### PR TITLE
FLUT-6128 - [Others] Provided the UG documentation to the filtering support.

### PR DESCRIPTION
**Pull request by ashok-kuvaraja**
## Description ##

* We have provided the UG document to the new data grid filtering feature.

### CODE REVIEW

File: Flutter/datagrid/filtering.md [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+            );
+          }),
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## UI Filtering
```
Comment (NeelakandanKannan) [inline Flutter/datagrid/filtering.md:370]: @ashok-kuvaraja  - Add the following topic above to this topic,
**Retrieving the filtered rows**
https://help.syncfusion.com/windowsforms/datagrid/filtering#retrieving-the-filtered-records

Then, add below topic,

**Filter DateColumn with range between two dates**
https://help.syncfusion.com/windowsforms/datagrid/filtering#filter-datecolumn-with-range-between-two-dates

**Review by NeelakandanKannan — CHANGES_REQUESTED** [review]
I have added my feedback as comments. Get back to me with the changes.

File: Flutter/datagrid/filtering.md [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+                  padding: EdgeInsets.symmetric(horizontal: 16.0),
+                  alignment: Alignment.centerRight,
+                  child: Text(
+                    'Salary',
+                    overflow: TextOverflow.ellipsis,
+                  ))),
+        ]),
+      ),
+      MaterialButton(
+          child: Text('Add Filter'),
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:136]: Modified.

File: Flutter/datagrid/filtering.md [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:200]: Modified

File: Flutter/datagrid/filtering.md [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+          onPressed: () {
+            // To clear filter conditions from a specific column.
+            _employeeDataSource.clearFilters(columnName: 'Salary');
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:203]: Modified

File: Flutter/datagrid/filtering.md [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Filter behavior
+
+The `FilterBehavior` enum property is used to specify whether to consider the `FilterValue` as the string or actual data type.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:217]: Modified

File: Flutter/datagrid/filtering.md [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Filter operator
+
+The `FilterOperator` enum property is used to decide how a logical operator is to be applied between multiple filter conditions.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:290]: Modified

File: Flutter/datagrid/filtering.md [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+    ],
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## UI Filtering
+
+The `SfDataGrid` provides excel like filtering UI and also advanced filter UI to filter the data easily. UI filtering can be enabled by setting `SfDataGrid.allowFiltering` property to `true`. This allows to open the filter UI by clicking on the filter icon in column header to filter the records. The filtering UI will be shown as a popup menu on the desktop and web platforms and it will be shown as a new page on the mobile platforms.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:372]: Modified

File: Flutter/datagrid/filtering.md [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+
+The `SfDataGrid` provides the following types of filter popup modes,
+
+* **Checkbox Filter** - Provides excel like filter interface with list of check boxes.
+* **Advanced Filter** - Provides advanced filter options to filter the data.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:377]: Modified

File: Flutter/datagrid/filtering.md [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+
+![Flutter datagrid shows a advanced filter in web platform](images/filtering/flutter-datagrid-advanced-filter-view.png)
+
+![Flutter datagrid shows a checkbox filter in mobile platform](images/filtering/flutter-datagrid-checkbox-filter-view-mobile.png)
+
+![Flutter datagrid shows a advanced filter in mobile platform](images/filtering/flutter-datagrid-advanced-filter-view-mobile.png)
+
+### Checkbox filtering
+
+The Checkbox filtering is like the Excel like filter popup that shows the checked list box of the unique items with the search text field. The items which are in the checked state will be visible in the view other items will be filtered out from the view.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:389]: Modified

File: Flutter/datagrid/filtering.md [hunk 10]
```diff
@@ -0,0 +1,616 @@
...
+* **Date Filters** – Loads various menu options and `DatePicker` to filter DateTime type column.
+
+<table>
+<tr>
+<th>Text Filters </th>
+<th>Number Filters </th>
+<th>Date Filters </th>
+</tr>
+<tr>
+<td>When the string value is bounded to the <code>GridColumn</code>, then <code>TextFilters</code> are loaded in <code>AdvancedFilterControl</code>. </td>
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:411]: Modified

File: Flutter/datagrid/filtering.md [hunk 11]
```diff
@@ -0,0 +1,616 @@
...
+<td>When the int or double values are bounded to the <code>GridColumn</code>, then <code>NumberFilters</code> are loaded in <code>AdvancedFilterControl</code>. </td>
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:412]: Modified

File: Flutter/datagrid/filtering.md [hunk 12]
```diff
@@ -0,0 +1,616 @@
...
+<td> When the DateTime type value is bounded to the <code>GridColumn</code>, then <code>DateFilters</code> are loaded in <code>AdvancedFilterControl</code>. </td>
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:413]: Modified

File: Flutter/datagrid/filtering.md [hunk 13]
```diff
@@ -0,0 +1,616 @@
...
+</tr>
+<table>
+
+![Flutter datagrid shows a advanced filter popup view](images/filtering/flutter-datagrid-advanced-view-before-filter.png)
+
+![Flutter datagrid shows the filtered rows](images/filtering/flutter-datagrid-advanced-view-after-filter.png)
+
+#### Case sensitive filtering
+
+The case sensitive filtering can be enabled for the column by using the casing icon available in the advanced filter UI. If the icon is active, the filtering will be applied with the case sensitive with the filter text. The case sensitive icon will be shown only to the string type columns.
```
[the remark this answers is not in the document]
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:433]: Modified

File: Flutter/datagrid/filtering.md (hunk 1 above)
Comment (ashok-kuvaraja) [reply Flutter/datagrid/filtering.md:370]: Included the topics.

**Review by NeelakandanKannan — APPROVED** [review]
The changes are fine.

**Pull request merged**